src/OrbitGl/CallstackThreadBar.cpp

```diff
@@ -125,11 +126,11 @@ void CallstackThreadBar::DoUpdatePrimitives(PrimitiveAssembler& primitive_assemb
     };
 
     if (GetThreadId() == orbit_base::kAllProcessThreadsTid) {
-      capture_data_->GetCallstackData().ForEachCallstackEventInTimeRange(
-          min_tick, max_tick, action_on_callstack_events);
+      capture_data_->GetCallstackData().ForEachCallstackEventInTimeRangeDiscretized(
+          min_tick, max_tick, resolution_in_pixels, action_on_callstack_events);
     } else {
-      capture_data_->GetCallstackData().ForEachCallstackEventOfTidInTimeRange(
-          GetThreadId(), min_tick, max_tick, action_on_callstack_events);
+      capture_data_->GetCallstackData().ForEachCallstackEventOfTidInTimeRangeDiscretized(
+          GetThreadId(), min_tick, max_tick, resolution_in_pixels, action_on_callstack_events);
     }
 
     // Draw selected callstack samples.
```

Comment:
What about the selected callstacks? Are we not using the same technique for those? I would imagine that's also performance-sensitive with a large enough selection.

Reply:
Maybe this is a case where it's worth keeping both paths - if the # of selected callstacks is larger than the screen resolution, use the optimization, otherwise just go through all of them?

Reply:
Thanks Davide. I missed them.

I don't like keeping both paths, the new one should be good enough in all cases, since the optimization is skipping empty pixels. There is a tiny regression when the average is up to 4 callstack per pixel but not noticeable.

Reply:
Sounds good to me.